sqlglot/dialects/spark.py

```diff
@@ -41,6 +41,17 @@ class Parser(Spark2.Parser):
         }
 
     class Generator(Spark2.Generator):
+        TYPE_MAPPING = {
+            **Spark2.Generator.TYPE_MAPPING,
+            exp.DataType.Type.TINYINT: "TINYINT",
+            exp.DataType.Type.SMALLINT: "SMALLINT",
+            exp.DataType.Type.BIGINT: "BIGINT",
+            exp.DataType.Type.MONEY: "DECIMAL(15, 4)",
+            exp.DataType.Type.SMALLMONEY: "DECIMAL(6, 4)",
```

Comment:
how did you figure out this mapping?

Reply:
Not sure if you're asking about the `MONEY` types Toby, but I'm leaving this for reference, regarding the `INT` types:

```
Time taken: 4.181 seconds, Fetched 1 row(s)
spark-sql (default)> select cast(1 as byte);
CAST(1 AS TINYINT)
1
Time taken: 0.234 seconds, Fetched 1 row(s)
spark-sql (default)> select cast(1 as short);
CAST(1 AS SMALLINT)
1
Time taken: 0.229 seconds, Fetched 1 row(s)
spark-sql (default)> select cast(1 as long);
CAST(1 AS BIGINT)
1
Time taken: 0.207 seconds, Fetched 1 row(s)
```

Seems like Spark 3 uses `TINYINT`, `SMALLINT` and `BIGINT` instead of `BYTE`, `SHORT` and `LONG`, respectively, which are the current types as defined in the Spark2 dialect.

Reply:
It looks like people [consider](https://stackoverflow.com/questions/582797/should-you-choose-the-money-or-decimalx-y-datatypes-in-sql-server) `DECIMAL(19, 4)` and `DECIMAL(10, 4)` for `MONEY` and `SMALLMONEY` accordingly.